best-practices/native/react-native/common-components/phone-number-test.js

```diff
@@ -0,0 +1,27 @@
+// example jest test for phone number component
+import React from 'react';
+import { Text } from 'react-native';
+import PhoneNumber from './phone-number';
+
+import renderer from 'react-test-renderer';
+
+it('renders correctly', () => {
+  const tree = renderer.create(
+    <PhoneNumber
+      phoneNo="555-1234"
+      disallowFontScaling
+    />
+    ).toJSON();
+  expect(tree).toMatchSnapshot();
+  
+  // only one child (should be a text)
+  expect(tree.children.length).toBe(1);
+
+  const text = tree.children[0];
+  // make sure explicit and default props are as expected
+  expect(text.children[0]).toBe('555-1234');
+  expect(text.props.style[1].fontSize).toBe(12);
+  expect(text.props.allowFontScaling).toBe(false);
```

Comment:
Thanks for adding the tests! 

The below is 100% optional.

When you are testing objects/components, you can add context within your tests to better split up your tests. You can wrap them in `describes` that will give your tests a bit more information to those attempting to debug your code. It literally describes the component (in this case `PhoneNumber`) and lets you create context in which your component is testing for. For example, in your test above, you could add 
```javascript
describe('PhoneNumber', function() {
  describe('when phoneNo is defined', function() {
      describe('renders correctly', () => {
      it('is expected to have a child with 555-1234') => {
        expect block here....
      });
  });
});
```

Ill often do the following in jest to make it look a little bit nice and make it explicit what I am testing:
```javascript
const context = describe;

describe('PhoneNumber', function() {
  describe('render()', function() {
      context('when phoneNo is defined', () => {
      it('is expected to have a child with 555-1234') => {
        expect block here....
      });
  });
});
```
In this case, the describe blocks help with describing the component and its method we are testing, whereas the context block helps us by specifying what inputs we are bringing to the test. Another option for you if you want to go down that path.

There is a lot of discussion on BDD with big it blocks VS wrapping `describes` for years. A good article: https://www.bignerdranch.com/blog/why-do-javascript-test-frameworks-use-describe-and-beforeeach/